udata/rdf.py

```diff
@@ -301,22 +302,64 @@ def contact_point_from_rdf(rdf, dataset):
             or rdf_value(contact_point, VCARD.email)
             or rdf_value(contact_point, DCAT.email)
         )
-        if not email:
+        email = email.replace("mailto:", "").strip() if email else None
+        contact_form = rdf_value(contact_point, VCARD.hasUrl)
+        if not email and not contact_form:
             return
-        email = email.replace("mailto:", "").strip()
         if dataset.organization:
             contact_point = ContactPoint.objects(
-                name=name, email=email, organization=dataset.organization
+                name=name, email=email, contact_form=contact_form, organization=dataset.organization
             ).first()
             return (
                 contact_point
-                or ContactPoint(name=name, email=email, organization=dataset.organization).save()
+                or ContactPoint(
+                    name=name,
+                    email=email,
+                    contact_form=contact_form,
+                    organization=dataset.organization,
+                ).save()
```

Comment:
Nit: would it make sense to use `udata.mongo.queryset.UDataQuerySet.get_or_create` for such use cases here and below with the owner?

Reply:
:+1:  Indeed! Done in https://github.com/opendatateam/udata/pull/3179/commits/75091ce7d009fa5049ca827eefb77f375933169b